community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java

```diff
@@ -125,6 +130,7 @@ public AllStoreHolder( StorageEngine engine,
         this.properties = statement.properties();
         this.explicitIndexStore = explicitIndexStore;
         this.procedures = procedures;
+        this.schemaState = engine.schemaState();
```

Comment:
as already mentioned storage engine should not be used as schema state provider.